engine/src/main/java/org/terasology/logic/console/commands/CoreCommands.java

```diff
@@ -461,12 +459,30 @@ public String leave() {
      * @throws IOException thrown when error with saving file occures
      */
     @Command(shortDescription = "Writes out information on all entities to a text file for debugging",
-            helpText = "Writes entity information out into a file named \"entityDump.txt\".")
-    public void dumpEntities() throws IOException {
+            helpText = "Writes entity information out into a file named \"entityDump.txt\"." +
+                    " Supports list of component names, which will be used to only save entities that contains" +
+                    " all or some of those components. List of component names must be provided in double quotes" +
+                    " and all names must be separated by spaces.")
+    public void dumpEntities(@CommandParam(value = "componentNames", required = false) String componentNames) throws IOException {
         EngineEntityManager engineEntityManager = (EngineEntityManager) entityManager;
         PrefabSerializer prefabSerializer = new PrefabSerializer(engineEntityManager.getComponentLibrary(), engineEntityManager.getTypeSerializerLibrary());
         WorldDumper worldDumper = new WorldDumper(engineEntityManager, prefabSerializer);
-        worldDumper.save(PathManager.getInstance().getHomePath().resolve("entityDump.txt"));
+        if (componentNames == null) {
+            worldDumper.save(PathManager.getInstance().getHomePath().resolve("entityDump.txt"));
+        } else {
+            Reflections reflections = new Reflections("org.terasology");
```

Comment:
Please avoid building your own `Reflections` object here. You can retrieve a `org.terasology.module.ModuleEnvironment` (it's a gestalt-module class, not in the engine) from the `ModuleManager` (using `ModuleManager.getEnvironment`) in the context and use that instead.

`ModuleEnvironment` already contains a `Reflections` object (and wrapper methods that delegate to that object) built over the engine and all loaded modules, so building the new `Reflections` object here is slow and completely unnecessary, and also may not encompass all loaded modules.

Reply:
Thanks, will do.

Reply:
Done

Reply:
👍